src/git.rs

```diff
@@ -79,40 +115,181 @@ impl Git for LiveGit {
+
+        let mut v = Vec::new();
+        let new_commit = self.repo.find_commit(to_id)?;
+        let mut current_id = to_id;
+        v.push(new_commit);
+        while current_id != from_id {
+            let current_commit = self.repo.find_commit(current_id)?;
+            for parent in current_commit.parents() {
+                current_id = parent.id();
+                let parent_commit = self.repo.find_commit(parent.id())?;
+                if current_id != from_id  {
+                    v.push(parent_commit);
+                }
+            }      
+        }
+        debug!("Commits found {:#?}",v);
+        Ok(v)         
+    }
+
+    fn find_unpushed_commits(&self, new_commit_id: Oid) -> Result<Vec<Commit<'_>>, Box<dyn Error>> {
+        debug!("Get unpushed commits from {} ", new_commit_id);
+
+        let mut v = Vec::new();
+        let new_commit = self.repo.find_commit(new_commit_id)?;
+        let mut current_id = new_commit_id;
+        let mut parent_count = new_commit.parent_count();
+        let mut pushed = self.pushed(current_id)?;
+        if !pushed {
+            v.push(new_commit);
         }
-        fn read_file(&self, _path: &str) -> Result<String, Box<dyn Error>> {
-            Ok(String::from(""))
+        while !pushed && parent_count > 0 {
+            let current_commit = self.repo.find_commit(current_id)?;
+            parent_count = current_commit.parent_count();
+            for parent in current_commit.parents() {
+                current_id = parent.id();
+                let parent_commit = self.repo.find_commit(parent.id())?;
+                pushed = self.pushed(current_id)?;
+                debug!("Commit {} already pushed? {}", current_id, pushed);
+                if !pushed  {
+                    v.push(parent_commit);
+                }
+            }      
         }
-        fn current_branch(&self) -> Result<String, Box<dyn Error>> {
-            Ok(String::from("master"))
+        debug!("Unpushed Commits found {:#?}",v);
+        Ok(v)         
+    }
+
+    fn verify_commit_signature(&self, commit_id: Oid) -> Result<String, Box<dyn Error>> {
+        debug!("Verify commit {}", commit_id);
+        let repo_path = self.repo.path();
+        debug!("Repo path {:?}", repo_path);
+        let result = Command::new("git")
+            .current_dir(repo_path)
+            .arg("verify-commit")
+            .arg("--raw")
+            .arg(commit_id.to_string())
+            .output()?;
+        debug!("RESULT {:?}", result);
+        if !result.status.success() {
+            return Err(Box::new(CapnError::new(format!("Call to git verify failed for commit {} with status {}", commit_id,result.status))));
         }
-        fn write_git_file(&self, _path: &str, _file_mode: u32, _contents: &str) -> Result<(), Box<dyn Error>> {
-            Ok(())
+
+        let encoded = String::from_utf8(result.stderr)?;
+        let fingerprints = encoded.split('\n')
```

Comment:
Isn't there only one fingerprint here?
We could call first after collect instead of binding a second variable.